pulser/waveforms.py

```diff
@@ -538,13 +538,27 @@ def from_max_val(
             raise ValueError(
                 "The maximum value and the area must have " "matching signs."
             )
+
         # A normalized Blackman waveform has an area of 0.42 * duration
         duration = np.ceil(area / (0.42 * max_val) * 1e3)  # in ns
         wf = cls(duration, area)
+
         # Adjust for rounding errors to make sure max_val is not surpassed
         while np.abs(wf._scaling) > np.abs(max_val):
             duration += 1
+            previous_wf = wf
             wf = cls(duration, area)
+
+        # According to the documentation for numpy.blackman(), "the value one
+        # appears only if the number of samples is odd". Hence, the previous
+        # even duration can reach a maximal value closer to max_val.
+        if (
+            previous_wf is not None
```

Comment:
I think you have to initialize `previous_wf = None` to make this work. Also, maybe add a test that covers the case where the `while` loop is not even entered.

Reply:
> I think you have to initialize `previous_wf = None` to make this work.

It worked though. The variable is declared above but never initialized, so it's set to None by default. But I will add it for clarity/safety.

> Also, maybe add a test that covers the case where the `while` loop is not even entered.

I assume this is already achieved by the existing tests. Plus, this part of code is not impacted by/related to this PR.

Reply:
Thanks for adding it explicitly